community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/pipes/CountStoreRelationshipAggregationPipe.scala

```diff
@@ -0,0 +1,53 @@
+package org.neo4j.cypher.internal.compiler.v2_3.pipes
+
+import org.neo4j.cypher.internal.compiler.v2_3.ExecutionContext
+import org.neo4j.cypher.internal.compiler.v2_3.executionplan.Effects
+import org.neo4j.cypher.internal.compiler.v2_3.planDescription.{NoChildren, PlanDescriptionImpl}
+import org.neo4j.cypher.internal.compiler.v2_3.symbols.SymbolTable
+import org.neo4j.cypher.internal.frontend.v2_3.NameId
+import org.neo4j.cypher.internal.frontend.v2_3.ast.RelTypeName
+import org.neo4j.cypher.internal.frontend.v2_3.symbols._
+
+case class CountStoreRelationshipAggregationPipe(ident: String, startLabelName: Option[String],
+                                                 typeNames: Seq[RelTypeName], endLabelName: Option[String])
+                                                (val estimatedCardinality: Option[Double] = None)
+                                                (implicit pipeMonitor: PipeMonitor) extends Pipe with RonjaPipe {
+
+  protected def internalCreateResults(state: QueryState): Iterator[ExecutionContext] = {
+    val baseContext = state.initialContext.getOrElse(ExecutionContext.empty)
+    val labelIds = Seq(startLabelName, endLabelName).map { labelName =>
```

Comment:
having `labelIds(0) == id of start label` and `labelIds(1) == id of end label` makes the code harder to read. Store them as separate variables (and don't do lookups for each row)